plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/preferences/SSLCertificatesPreference.java

```diff
@@ -105,13 +106,19 @@ public void setAllowedByCertificate(X509Certificate certificate, boolean result)
 	private boolean isValid(String validity) {
 		int expiresOnIndex = validity.indexOf(X509CertificateParser.EXPIRES_ON_PREFIX);
 		if(expiresOnIndex >= 0) {
-			SimpleDateFormat dateFormat = new SimpleDateFormat(X509CertificateParser.DATE_FORMAT);
+			SimpleDateFormat dateFormat = new SimpleDateFormat(X509CertificateParser.DATE_FORMAT,  Locale.ENGLISH);
```

Comment:
Maybe test like this will help

package org.jboss.tools.openshift.test.common.ui.utils;

import java.security.cert.X509Certificate;
import java.util.Date;
import java.util.Locale;

import org.jboss.tools.openshift.common.core.utils.X509CertificateParser;
import org.junit.Assert;
import org.junit.Test;
import org.mockito.Mockito;

public class X509CertificateParserTest {
	
	@Test
	public void testValidity() {
		X509Certificate certificate = Mockito.mock(X509Certificate.class);
		Mockito.when(certificate.getNotBefore()).thenReturn(new Date(new Date().getTime() - 1000000));
		Mockito.when(certificate.getNotAfter()).thenReturn(new Date(new Date().getTime() - 1000000));
		String validity = X509CertificateParser.getValidity(certificate, Locale.ENGLISH);
		Assert.assertFalse("Date is not valid, true means that validation failed with parser error.", X509CertificateParser.isValid(validity));
		
		//Set unlikely locale 
		validity = X509CertificateParser.getValidity(certificate, Locale.TAIWAN);
		Assert.assertTrue("Date is not valid, but locale is wrong, parser error is expected.", X509CertificateParser.isValid(validity));

		validity = X509CertificateParser.getValidity(certificate, Locale.getDefault());
		System.out.println("---------------->" + Locale.getDefault());
		Assert.assertFalse("Date is not valid, true means that validation failed with parser error.", X509CertificateParser.isValid(validity));
	}

}